nodejs/eks/userdata.ts

```diff
@@ -211,6 +215,102 @@ ${cfnSignal}
 `;
 }
 
+function createNodeadmUserData(clusterMetadata: ClusterMetadata, args: UserDataArgs): string {
+    if (args.bootstrapExtraArgs) {
+        throw new pulumi.ResourceError(
+            "The 'bootstrapExtraArgs' argument is not supported for nodeadm based user data.",
+            undefined,
+        );
+    }
+
+    const nodeadmSkeleton = {
+        apiVersion: "node.eks.aws/v1alpha1",
+        kind: "NodeConfig",
+    };
+
+    const baseConfig = {
+        ...nodeadmSkeleton,
+        spec: {
+            cluster: {
+                name: clusterMetadata.name,
+                apiServerEndpoint: clusterMetadata.apiServerEndpoint,
+                certificateAuthority: clusterMetadata.certificateAuthority,
+                cidr: clusterMetadata.serviceCidr,
+            },
+        },
+    };
+
+    const nodeadmPrefix = "---\n";
+
+    const parts = [
+        {
+            contentType: "application/node.eks.aws",
+            content: nodeadmPrefix + jsyaml.dump(baseConfig),
+        },
+    ];
+
+    // nodeadm config gets iteratively merged together, so we can add a new section for the kubelet flags
+    const kubeletFlags = buildKubeletFlags(args);
+    if (kubeletFlags.length > 0) {
+        parts.push({
+            contentType: "application/node.eks.aws",
+            content:
+                nodeadmPrefix +
+                jsyaml.dump({
+                    ...nodeadmSkeleton,
+                    spec: {
+                        kubelet: {
+                            flags: kubeletFlags,
+                        },
+                    },
+                }),
+        });
+    }
+
+    // TODO: expose extra nodeadm config options in the schema
```

Comment:
Ticket tracking this or this is done now? Or not needed?

Reply:
It's captured in the parent ticket. I just want to tackle it in a separate PR not to blow this one up